### PR TITLE
[cases](array)first commit for array cases

### PR DESCRIPTION
**Pull request by amorynan**
### What problem does this PR solve?
this is some special data case for array , and main to use in function and ut for making up the array cases 
Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

File: regression-test/data/nereids_function_p0/array/test_array_char(255).csv [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// this is data case file for array<varchar|string> type, contains normal, corner, exceptional data cases
+[]
+[null]
+NULL
+// single element
+['0']
```
Comment (aytrack) [inline regression-test/data/nereids_function_p0/array/test_array_char(255).csv:6]: Empty string: ""

File: regression-test/data/nereids_function_p0/array/test_array_array_bigint.csv [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[[]]
+[[null], [null], [null], [null], [null], [null], [null], [null], [null], [null]]
+NULL
+[[0], [0], [0], [0], [0], [0], [0], [0], [0], [0]]
+[[-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807], [-9223372036854775808, 9223372036854775807]]
+[[-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808], [-9223372036854775807, 9223372036854775808]]
```
Comment (aytrack) [inline regression-test/data/nereids_function_p0/array/test_array_array_bigint.csv:6]: how about add a array like `[[], null, [null], 1]` which contains null, normal data and zero value

File: regression-test/data/nereids_function_p0/scalar_function/Array1.out
Comment (eldenmoon) [inline regression-test/data/nereids_function_p0/scalar_function/Array1.out:1]: how can you make sure the large output is correct

File: regression-test/data/nereids_function_p0/array/test_array_map.csv [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// this is data case file for array<map<largeInt:string,datetime:decimal<76,56>, char:double, ipv4:ipv6>> type, contains normal, corner, exceptional data cases
```
Comment (eldenmoon) [inline regression-test/data/nereids_function_p0/array/test_array_map.csv:1]: what about array<json> case

File: regression-test/suites/nereids_function_p0/load.groovy [hunk 4]
```diff
@@ -556,6 +558,267 @@ suite("load") {
...
+                    col_content.add(line)
+                }
+                max_row = Math.max(max_row, col_content.size())
+                table_content.add(col_content)
+            } catch (IOException e) {
+                e.printStackTrace()
+            }
+        }
+        // construct insert sql
+        for (int i = 0; i < max_row; i++) {
```
Comment (eldenmoon) [inline regression-test/suites/nereids_function_p0/load.groovy:722]: what about seperate to serveral batches of insert? and compaction will be tested

File: regression-test/data/nereids_function_p0/scalar_function/Array1.out
Comment (amorynan) [reply regression-test/data/nereids_function_p0/scalar_function/Array1.out:1]: to be honest, sql will not make coredump and throw not expected exception is the most confirmatory result, and result correctness really needs to be seen with the naked eye

File: regression-test/suites/nereids_function_p0/load.groovy (hunk 4 above)
Comment (amorynan) [reply regression-test/suites/nereids_function_p0/load.groovy:722]: do we have group commit ？

File: regression-test/data/nereids_function_p0/array/test_array_map.csv (hunk 3 above)
Comment (amorynan) [reply regression-test/data/nereids_function_p0/array/test_array_map.csv:1]: we do not support create data type array nested json

File: regression-test/data/nereids_function_p0/array/test_array_char(255).csv (hunk 1 above)
Comment (amorynan) [reply regression-test/data/nereids_function_p0/array/test_array_char(255).csv:6]: done!

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Review by py023 — APPROVED** [review]

**Comment by htyoung** [comment]
Since Windows does not support the use of < and > symbols in file names, can the file name in this PR be changed so that git can be used to successfully pull the code on Windows?

**Pull request merged**